### DISCUSSION TITLE
TypeError:  **** is not a function

### DISCUSSION THREAD

**Discussion by denganjia** [vitest-dev/vitest#6471] [Q&A]
我使用React + Zustand，并且使用了[Slices pattern](https://zustand.docs.pmnd.rs/guides/typescript#slices-pattern)。当我运行测试时，会提示TypeError: **** not a function。我保证这个函数绝对是存在且有效的，这个问题困扰了我好几天了，导致我现在关于zustand store部分的测试都不能运行。
这里是一个最小的[运行示例](https://stackblitz.com/edit/vitejs-vite-dvoxd2?file=src%2Fstore%2Fmodules%2FaSlice.ts)

I use React + Zustand, and I use [Slices pattern](https://zustand.docs.pmnd.rs/guides/typescript#slices-pattern). When I run the test, I get TypeError: **** not a function. I guarantee that this function absolutely exists and works, and this problem has been bothering me for several days, causing my current tests on zustand store to fail.
Here is a [minimal example](https://stackblitz.com/edit/vitejs-vite-dvoxd2?file=src%2Fstore%2Fmodules%2FaSlice.ts) in action

**Comment by AriPerkkio** [comment]
There are plenty of circular dependencies caused by those files. They all end up importing each other. I would recommend to try to refactor the code so that `aSlice.ts` doesn't import `index.ts`, and `createHook.ts` wouldn't import `index.ts`.

**Reply by denganjia** [reply]
Perhaps there is a way for vitest to ignore circular references? Load everything in a file or scope

**Reply by AriPerkkio** [reply]
There's an open issue for that: https://github.com/vitest-dev/vitest/issues/4143
